services/ui_backend_service/api/dag.py

```diff
@@ -85,6 +72,68 @@ async def get_run_dag(self, request):
 
         return web_response(status, body)
 
+    async def get_run_ids_from_codepackage_metadata(self, request):
+        """
+        Tries to locate 'code-package' or 'code-package-url' in run metadata.
+
+        Returns
+        -------
+        Tuple
+            flow_id : str
+            run_id : str or int
+        """
+        flow_name = request.match_info['flow_id']
+        run_id_key, run_id_value = translate_run_key(
+            request.match_info.get("run_number"))
+        # 'code-package' value contains json with dstype, sha1 hash and location
+        # 'code-package-url' value contains only location as a string
+        db_response, *_ = await self.db.metadata_table_postgres.find_records(
+            conditions=[
+                "flow_id = %s",
+                "{run_id_key} = %s".format(
+                    run_id_key=run_id_key),
+                "(field_name = %s OR field_name = %s)"
+            ],
+            values=[
+                flow_name, run_id_value,
+                "code-package", "code-package-url"
+            ],
+            fetch_single=True, expanded=True
+        )
+
+        return db_response
+
+    async def get_run_ids_from_graph_info_artifact(self, request):
+        """
+        Tries to locate '_graph_info' in run artifacts
+
+        Returns
+        -------
+        Tuple
+            flow_id : str
+            run_id : str or int
+        """
```

Comment:
doesn't seem to return a tuple here either

Reply:
cleaned up dag api router a bit, including the type notation.